SalesforceCommerceCloud/pwa-kit
# Hooks integration - My Account Order History

# Description


The PR updates the My Account Order History to use commerce-sdk-react hooks.

# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- Removed Order context.
- Updated Order history and Order details to use commerce-sdk-react hooks.

# How to Test-Drive This PR
1. Checkout this branch `hooks-checkout-order-summary`
2. Start the `template-retail-react` project: 
```
npm run start --prefix packages/template-retail-react-app/
```
3. Login with a registered user account with multiple orders already placed.
4. Navigate to My Account > Order History. http://localhost:3000/global/en-GB/account/orders
    **Confirm:**
    - The expected orders are rendered.
    - The pagination works as expected. Play with the `DEFAULT_PAGINATION_LIMIT` value currently set to 10 if your registered user has less than 10 orders to see the pagination.
    https://github.com/SalesforceCommerceCloud/pwa-kit/blob/a28cb3625ec52926e240b36e63ff5376320e26d6/packages/template-retail-react-app/app/pages/account/order-history.jsx#L35
 
 5. Click on any of the orders listed to navigate to Order detail and verify the expected content is rendered. 
 

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
